content/blog/cruxbox/2020-08-06_gsoc_2020_project_xfs_project_update.md

```diff
@@ -0,0 +1,46 @@
++++
+type = "blog"
+title = "GSoC 2020 Project Update: XFS"
+author = "CruxBox"
+date = "2020-11-08 17:58:00+05:30"
+tags = ["haiku", "software", "gsoc", "gsoc2020", "file system"]
++++
+
+Hello there!
+There is quite a lot to share about the work done this month.
+---
+
+Work done so far:
+You can check for the patches here: https://review.haiku-os.org/q/owner:shubhambhagat111%2540yahoo.com
+In short: All directory work is done. This includes ShortForm Directories, Block Directories, Leaf Directories, Node Directories and B+Tree based Directories.
+
+Detailed view at the work:
+Some context: Leaf entries containt hash values of a directory entry name and also the address of that entry. We use this for fast lookups.
+1) Leaf Directories:
+
+When a directory block (directory blocks are larger than normal file system blocks) isn't enough to hold all directory entries and the leaf entries within a single directory block, the block expands and now the directory entries and leaf entries stay entirely in different directory blocks. There could be many directory blocks but there is only a single leaf block. All block mappings are present within the inode itself. I made sure to use binary search (with a modified predicate ofcourse) within the leaf block to reduce the time complexity of Lookup(). It was interesting to see how different structures play here. The original docs were quite helpful.
```

Comment:
Maybe find a good place to break up this long paragraph.

Reply:
also "ofcourse" -> "of course"

Reply:
Ack